packages/main/src/ListItem.ts

```diff
@@ -293,10 +293,6 @@ abstract class ListItem extends ListItemBase {
 		const itemActive = this.type === ListItemType.Active,
 			itemNavigated = this.typeNavigation;
 
-		if (isSpace(e)) {
```

Comment:
Are we sure that this applies to all items, because I remember that we wanted to prevent this for some reason. 
So if have items that needs it and items that don't need it, we probably have to move these code wherever needed respectively. But if all classes (extending ListItem) do not need it - it's fine, probably @dobrinyonkov would know as a11y expert.

Reply:
This code's purpose is to prevent the list from scrolling when spacebar is pressed on a list item. I guess we can move it to the List's onkeydown handler and check also if `e.target instanceof ListItemBase`. This should effectively achieve the same ensuring that pressing space while a list item is focused won't trigger scrolling, as well as retain the spacebar functionality while typing in an input.

A side effect I would expect is that interactive elements nested within list items, which do not themselves prevent this default behavior, may still unintentionally trigger list scrolling when the spacebar is pressed on them. However, the list does this even with its implmentation now, so I guess we could rely on the above.

For instance, the following custom list item content behaves the same with or without the new additional check `e.target instanceof ListItemBase` in the `List.ts`

```html
	<ui5-li-custom>
		<div tabindex="0">press space on me</div>
		<button>press space on me</button>
		<ui5-button>press space on me</ui5-button>
		<input placeholder="press space on me" type="text">
		<ui5-input placeholder="press space on me"></ui5-input>
	</ui5-li-custom>
```

Reply:
I moved the prevention in the List.ts as suggested.

There is a slight side effect where if we press space in the TreeItemCustom in Input (the issue we are fixing) we now have proper space character but the item is also toggled. Do you think this should be addressed also ?

Reply:
Great finding. Thanks, it works flawlessly.